test/binder/binder_test.cpp

```diff
@@ -0,0 +1,820 @@
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "binder/bind_node_visitor.h"
+#include "catalog/catalog.h"
+#include "parser/expression/aggregate_expression.h"
+#include "parser/expression/column_value_expression.h"
+#include "parser/expression/operator_expression.h"
+#include "parser/expression/subquery_expression.h"
+#include "parser/postgresparser.h"
+#include "storage/garbage_collector.h"
+#include "traffic_cop/statement.h"
+#include "traffic_cop/traffic_cop.h"
+#include "transaction/deferred_action_manager.h"
+#include "transaction/transaction_manager.h"
+#include "util/data_table_benchmark_util.h"
+#include "util/test_harness.h"
+
+using std::make_tuple;
+
+using std::unique_ptr;
+using std::vector;
+
+namespace terrier {
+
+// TODO(Ling): write meaningful setup
+class BinderCorrectnessTest : public TerrierTest {
+ private:
+  storage::RecordBufferSegmentPool buffer_pool_{1000000, 1000000};
+  storage::BlockStore block_store_{1000, 1000};
+  catalog::Catalog *catalog_;
+  storage::GarbageCollector *gc_;
+
+ protected:
+  std::string default_database_name_ = "test_db";
+  catalog::db_oid_t db_oid_;
+  catalog::table_oid_t table_a_oid_;
+  catalog::table_oid_t table_b_oid_;
+  parser::PostgresParser parser_;
+  transaction::TimestampManager *timestamp_manager_;
+  transaction::DeferredActionManager *deferred_action_manager_;
+  transaction::TransactionManager *txn_manager_;
+  transaction::TransactionContext *txn_;
+  std::unique_ptr<catalog::CatalogAccessor> accessor_;
+  binder::BindNodeVisitor *binder_;
+
+  void SetUpTables() {
+    // Initialize the transaction manager and GC
+    timestamp_manager_ = new transaction::TimestampManager;
+    deferred_action_manager_ = new transaction::DeferredActionManager(timestamp_manager_);
+    txn_manager_ = new transaction::TransactionManager(timestamp_manager_, deferred_action_manager_, &buffer_pool_,
+                                                       true, DISABLED);
+    gc_ = new storage::GarbageCollector(timestamp_manager_, deferred_action_manager_, txn_manager_, nullptr);
+
+    // new catalog requires txn_manage and block_store as parameters
+    catalog_ = new catalog::Catalog(txn_manager_, &block_store_);
+
+    // create database
+    txn_ = txn_manager_->BeginTransaction();
+    LOG_INFO("Creating database %s", default_database_name_.c_str());
```

Comment:
Need to check with @mbutrovich + @lmwnshn on what the right log target to use here.